### PR TITLE
docs: add auth paragraph about oidc group sync

### PR DESCRIPTION
**Pull request by coadler**

### CODE REVIEW

File: docs/admin/auth.md [hunk 1]
```diff
@@ -120,18 +120,24 @@ Once complete, run `sudo service coder restart` to reboot Coder.
 
 ## OIDC Claims
 
-Coder requires all OIDC email addresses to be verified by default. If the `email_verified` claim is present in the token response from the identity provider, Coder will validate that its value is `true`.
-If needed, you can disable this behavior with the following setting:
+Coder requires all OIDC email addresses to be verified by default. If the
+`email_verified` claim is present in the token response from the identity
+provider, Coder will validate that its value is `true`. If needed, you can
+disable this behavior with the following setting:
```
Comment (bpmct) [inline docs/admin/auth.md:126]: TIL!

File: docs/admin/auth.md [hunk 2]
```diff
@@ -157,3 +163,22 @@ If your OpenID Connect provider requires client TLS certificates for authenticat
 CODER_TLS_CLIENT_CERT_FILE=/path/to/cert.pem
 CODER_TLS_CLIENT_KEY_FILE=/path/to/key.pem
 ```
+
+## Group Sync (enterprise)
+
+If your OpenID Connect provider supports group claims, you can configure Coder
+to synchronize groups in your auth provider to groups within Coder.
+
+To enable group sync, ensure that the `group` claim is set:
```
Comment (bpmct) [inline docs/admin/auth.md:172]: Do some OIDC providers send this claim by default or is this always necessary on the Coder side? Wondering if a deployment might suddenly start doing group sync without the admin realizing (and there happen to be some matching group names in the Coder deployment)?

**Review by bpmct — APPROVED** [review]
Nice! Just a minor question, but not something we necessarily need to include.

File: docs/admin/auth.md (hunk 2 above)
Comment (coadler) [reply docs/admin/auth.md:172]: As far as I can tell this is required on the coder side. All of the docs I read don't send group claims by default.

**Pull request merged**